global/scripts/CorrThick.sh

```diff
@@ -0,0 +1,181 @@
+    - Subject directory
+    - Subject list
+    - Location of the HCP Pipeline environment setup script (SetUpHCPPipeline.sh)
+
+Example batch script:
+---------------------------------------
+EnvironmentScript="/path/to/SetUpHCPPipeline.sh" # HCP Pipeline environment script
+source "\$EnvironmentScript"
+
+SubjectDir="/path/to/subjects"
+SubjList="100206 100307 100408 ...."  # space-separated subject IDs
+RegNames="MSMSulc@MSMAll"
+
+for Subject in \$SubjList; do
+    \"\$HCPPIPEDIR/global/scripts/CorrThick.sh\" \\
+        --subject-dir=\"\$SubjectDir\" \\
+        --subject=\"\$Subject\" \\
+        --regnames=\"\$RegNames\" \\
+        --skip-computation=NO
+done
+
+OR example usage with fsl_sub:
+
+for Subject in \$SubjList; do
+    fsl_sub -q matlabparallel.q \"\$HCPPIPEDIR/global/scripts/CorrThick.sh\" \\
+        --subject-dir=\"\$SubjectDir\" \\
+        --subject=\"\$Subject\" \\
+        --regnames=\"\$RegNames\"
+done
+---------------------------------------"
+
+opts_AddMandatory '--subject-dir' 'SubjectDir' 'path' "folder containing all subjects"
+opts_AddMandatory '--subject' 'Subject' 'subject ID' "subject-id"
+opts_AddOptional '--regnames' 'RegNamesStr' 'my reg str' "set the desired registration name(s) separated by @, 'string' 'RegName@RegName@RegName@...etc.' default MSMSulc" "MSMSulc"
+opts_AddOptional '--hemi' 'Hemi' 'hemisphere' "provide hemisphere for regression calculation, L=Left, R=Right, or B=Both, default 'B'" "B"
+opts_AddOptional '--surf' 'Surface' 'surface' "provide surface for regression calculation, white or midthickness, default midthickness" "midthickness"
+opts_AddOptional '--patch-size' 'PatchSize' 'distance' "provide patch kernel size in millimeters FWHM for regression, default 6" "6"
+opts_AddOptional '--surf-smooth' 'SurfSmooth' 'distance' "provide surface smoothing in millimeters FWHM, default 2.14" "2.14"
+opts_AddOptional '--metric-smooth' 'MetricSmooth' 'distance' "provide metric smoothing in millimeters FWHM, default 2.52" "2.52"
+opts_AddOptional '--skip-computation' 'SkipCompute' 'YES or NO' "whether or not to compute the curvature-corrected (folding-compensated) cortical thickness, if it is already available, but just to resample it to 164k and 32k, defaults to 'NO'" "NO"
+
+opts_ParseArguments "$@"
+
+if ((pipedirguessed))
+then
+	log_Err_Abort "HCPPIPEDIR is not set, you must first source your edited copy of Examples/Scripts/SetUpHCPPipeline.sh"
+fi
+
+#display the parsed/default values
+opts_ShowValues
+
+#sanity check boolean strings and convert to 1 and 0
+SkC=$(opts_StringToBool "$SkipCompute")
+
+#set paths
+NonlinearFolder="$SubjectDir"/"$Subject"/MNINonLinear
+NativeFolder="$NonlinearFolder"/Native
+T1wNativeFolder="$SubjectDir"/"$Subject"/T1w/Native
+
+#Make intermediate directory to save intermediate files
+mkdir -p "$NativeFolder"/CorrThick
+
+#Loop through left and right hemispheres
+if [[ "$Hemi" == *B* ]]
+then
+	Hemi="L R"
+fi
```

Comment:
Consider a more robust check that tells the user when they did something dumb:

```
case "$Hemi" in
    (L|R|"L R"|"R L")
        ;;
    (B)
        Hemi="L R"
        ;;
    (*)
        log_Err_Abort "unrecognized hemisphere specification '$Hemi', use L, R, or B"
        ;;
esac
```

This would also forbid "L L", "L B", and other silliness (the help info doesn't actually say "L R" is valid, either, so technically it could also be forbidden).  Most scripts don't have a way to disable hemispheres, so removing this option for simplicity is another possibility.